### PR TITLE
feat(design-system/TUX-1179/Tabs): New Tab components for DS

### PR DESCRIPTION
**Pull request by Benrajalu**
**What is the problem this PR is trying to solve?**
Tabs were previously composition-only and did not follow token guidelines. 

**What is the chosen solution to this problem?**
- Write an API-first Tabs component to handle most usecases
- Move current API (mostly) untouched in a composition-based kit 
- Create a "TabLinks" component for a series of links that look like tabs for Router jobs.

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

**Review by jmfrancois — APPROVED** [review]

File: packages/design-system/src/components/WIP/Tabs/Primitive/Tab.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+	TabSize &
+	TabTooltip &
+	TabChildren;
+
+type TabPropsTypes = TabPropsTypesWithoutState & TabState;
+
+const Tab = forwardRef((props: TabPropsTypes, ref: Ref<HTMLButtonElement>) => {
+	const { tooltip, ...otherProps } = props;
+
+	const component = () => {
```
Comment (sgangnant) [inline packages/design-system/src/components/WIP/Tabs/Primitive/Tab.tsx:41]: Do you know if there's an impact on performance between `component()` vs `<Component />` (the 2nd solution not dynamically declaring a component-like) ?
Just wondering, not world-changing point

File: packages/design-system/src/components/WIP/Tabs/variants/TabsAsLinkList.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+type TabsAsLinkListProps = {
+	tabs: TabAsLinkProps[];
+	size?: 'M' | 'L';
+} & DataAttributes;
+
+const TabsAsLinkList = forwardRef(
+	({ tabs, size = 'M', ...props }: TabsAsLinkListProps, ref: Ref<HTMLUListElement>) => {
+		return (
+			<div className={styles.tabList}>
+				<StackHorizontal gap={size} as={'ul'} {...props} ref={ref}>
```
Comment (sgangnant) [inline packages/design-system/src/components/WIP/Tabs/variants/TabsAsLinkList.tsx:17]:
```suggestion
				<StackHorizontal gap={size} as="ul" {...props} ref={ref}>
```

Comment (sgangnant) [reply packages/design-system/src/components/WIP/Tabs/variants/TabsAsLinkList.tsx:17]: I'm a linter

File: packages/design-system/src/components/WIP/Tabs/variants/TabsKit.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+export type TabsProps = {
+	children: ReactElement | ReactElement[];
+} & TabInitialState;
+
+const TabsContext = createContext<TabState | null>(null);
+
+const Tabs = ({ children, ...initialState }: TabsProps) => {
+	const tab = useTabState(initialState);
+	const value = useMemo(() => tab, Object.values(tab));
```
Comment (sgangnant) [inline packages/design-system/src/components/WIP/Tabs/variants/TabsKit.tsx:15]:
```suggestion
	const value = useMemo(tab, Object.values(tab));
```
Also, why not `[tab]` in itself as "cache identifier" ?

File: packages/design-system/src/components/WIP/Tabs/variants/TabsAsLinkList.tsx (hunk 2 above)
Comment (Benrajalu) [reply packages/design-system/src/components/WIP/Tabs/variants/TabsAsLinkList.tsx:17]: DAMN. Imagine if we had a linter.

File: packages/design-system/src/components/WIP/Tabs/variants/TabsKit.tsx (hunk 3 above)
Comment (Benrajalu) [reply packages/design-system/src/components/WIP/Tabs/variants/TabsKit.tsx:15]: Dunno, that's a carryover from frassinier's implem. I'll dig, thanks!

Comment (Benrajalu) [reply packages/design-system/src/components/WIP/Tabs/variants/TabsKit.tsx:15]: Screenshot 2022-07-21 at 10 49 28

But it seems like `[tab]` is a valid alternative.

**Review by tlnd-mhuchet — APPROVED** [review]

**Pull request merged**